lib/Parse/ParsePattern.cpp

```diff
@@ -242,26 +238,47 @@ Parser::parseParameterClause(SourceLoc &leftParenLoc,
       }
 
       // (':' ('inout')? type)?
+      // also handle the case of (':' ('let' or 'var')? type)? as errors
+      // and give diagnostics for that case
       if (consumeIf(tok::colon)) {
 
         SourceLoc postColonLoc = Tok.getLoc();
 
         bool hasDeprecatedInOut =
           param.SpecifierKind == ParsedParameter::InOut;
-
-        if (Tok.is(tok::kw_inout)) {
-          param.LetVarInOutLoc = consumeToken();
-          param.SpecifierKind = ParsedParameter::InOut;
-          if (hasDeprecatedInOut) {
-            diagnose(param.LetVarInOutLoc, diag::inout_as_attr_deprecated)
-              .fixItRemove(param.LetVarInOutLoc);
+        bool hasValidInOut = false;
+        
+        while (Tok.isAny(tok::kw_inout, tok::kw_let, tok::kw_var)) {
+          if (Tok.is(tok::kw_var)) {
+            diagnose(Tok.getLoc(), diag::parameter_var_as_attr)
+            .fixItRemove(Tok.getLoc());
+            consumeToken();
+            param.isInvalid = true;
+          } else if (Tok.is(tok::kw_let)) {
+            diagnose(Tok.getLoc(), diag::parameter_let_as_attr).
+            fixItRemove(Tok.getLoc());
```

Comment:
Do you think SE-0053 is highly likely to be accepted? If so it would be almost trivial to implement and merge in with this PR and also fix the diagnostics to be more uniform since both `var` and `let` will be disallowed.